### PR TITLE
VZ-4467 Change log levels in the auth lua to remove verbosity

### PR DESCRIPTION
**Pull request by davisgomes**
# Description

Currently, the auth.lua logger is emitting messages too quickly with little useful information. In response, I have downgraded all of the info messages to debugging. This is a first pass at fixing this issue. Future iterations will need to expand upon the error logging.

Fixes VZ-4467

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by markxnelson — APPROVED** [review]

**Review by markxnelson — DISMISSED** [review]

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 1]
```diff
@@ -773,16 +787,16 @@ data:
         if not publicKey then
             me.unauthorized("No public key found")
         end
-        -- me.info("TOKEN: iss is "..jwt_obj.payload.iss)
-        -- me.info("TOKEN: oidcIssuerUri is"..oidcIssuerUri)
+        me.debug("TOKEN: iss is "..jwt_obj.payload.iss)
```
Comment (kminder) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:790]: This probably shouldn't be logged for security reasons.  The token is sort of a credential.

Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:790]: Thanks. I will remove this.

**Review by wmhopkins — COMMENTED** [review]
Looks good to me. I don't think we should submit as is, though -- it will immediately make it impossible to debug any auth-proxy issue that occurs in testing unless we do something to enable debug for test runs.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 2]
```diff
@@ -773,16 +787,14 @@ data:
         if not publicKey then
             me.unauthorized("No public key found")
         end
-        -- me.info("TOKEN: iss is "..jwt_obj.payload.iss)
```
Comment (wmhopkins) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:776]: I think these are fine to leave in and convert to debug messages. It's just token issuer data, nothing sensitive.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ data:
     -- CORS handling
     local h, _ = ngx.req.get_headers()["origin"]
     if h ~= nil and h ~= "" then
-        auth.info("Origin header: "..h)
+        auth.debug("Origin header: "..h)
         if h == "*" then
             -- not a legit origin, could be intended to trick us into oversharing
             auth.bad_request("Invalid Origin header: '*'")
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:46]: Maybe debug

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 4]
```diff
@@ -94,10 +94,10 @@ data:
             token = auth.handleBasicAuth(authHeader)
         end
         if not token then
-            auth.info("No recognized credentials in authorization header")
+            auth.error("No recognized credentials in authorization header")
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:97]: debug

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 5]
```diff
@@ -514,21 +528,21 @@ data:
     -- should only be called if some vz process is trying to access vmi using basic auth
     -- tokens are cached locally
     function me.handleBasicAuth(authHeader)
-        -- me.info("Checking for basic auth credentials")
+        me.debug("Checking for basic auth credentials")
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:531]: Get rid of anything "About to do..." or "Checking this ..."

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 6]
```diff
@@ -677,7 +691,7 @@ data:
         end
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:691]: When talking with an external service, give context to what happened

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 7]
```diff
@@ -919,17 +931,17 @@ data:
         local cookie, err = require("resty.cookie"):new()
         local ck = cookie:get(ckName)
         if not ck then
-            me.info("Cookie not found")
+            me.error("Cookie not found")
             return nil
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:935]: debug

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 8]
```diff
@@ -919,17 +931,17 @@ data:
...
         local decoded = base64.decode_base64url(ck)
         if not decoded then
-            me.info("Cookie not decoded")
+            me.error("Cookie not decoded")
             return nil
         end
         local json = me.aes256:decrypt(decoded)
         if not json then
-            me.info("Cookie not decrypted")
+            me.error("Cookie not decrypted")
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:944]: Both debug

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 9]
```diff
@@ -952,7 +964,7 @@ data:
         end
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:964]: Look for external service call.
If so, log error.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 10]
```diff
@@ -952,7 +964,7 @@ data:
...
         local data = cjson.decode(res.body)
         if not (data.keys) then
-            me.info("No keys found")
+            me.error("No keys found")
```
Comment (davisgomes) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:967]: Check this for external call.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml (hunk 2 above)
Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:776]: I'm leaving all commented out messages as they are, but switching to debug.

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml [hunk 11]
```diff
@@ -947,12 +971,12 @@ data:
         local certsUri = me.getOidcCertsUri()
         local res, err = httpc:request_uri(certsUri)
         if err then
-            me.logJson(ngx.WARN, "Could not retrieve certs", err)
+            me.error("Could not retrieve certs"..err)
```
Comment (kminder) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:974]: This will probably format badly.

Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy-configmap.yaml:974]: Missed the colon. Thanks

**Review by kminder — DISMISSED** [review]
Approved.

**Review by wmhopkins — APPROVED** [review]
Looks good to me.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**